### PR TITLE
Taskset + security groups

### PR DESCRIPTION
**Pull request by glommer**
Tzach, I don't know if you received the first pull request. This here has two additional others.

I would like to send just the 2 other patches, but I don't think it is possible (unless I rebase or you merge the first one)

Let me know what do you want me to do.

### CODE REVIEW

**Comment by tzach** [comment]
Merged.
There is probably a cleaner way to do the loader CPU, irc.
I will open a ticket for it.

File: stress.yaml [hunk 1]
```diff
@@ -27,19 +27,32 @@
     - set_fact: index={{ec2_tag_index}}
       when: index is not defined
 
+    - set_fact: taskset="taskset -c {{load_cpus}}"
+      when: load_cpus is defined
```
Comment (tzach) [inline stress.yaml:31]: @glommer are you using different load_cpus in each run or is it a property of the loader server?

File: stress.yaml [hunk 2]
```diff
@@ -27,19 +27,32 @@
...
+
+    - set_fact: taskset=""
+      when: load_cpus is not defined
```
Comment (tzach) [inline stress.yaml:34]: Shouldn't we ALWAYS use load_cpus?

File: stress.yaml [hunk 3]
```diff
@@ -27,19 +27,32 @@
...
+
+    - shell: for irq in `cat  /proc/interrupts | grep eth0 | cut -d":" -f1`; do echo 1 > /proc/irq/$irq/smp_affinity; done
+      when: load_cpus is defined
+      sudo: yes
+
     - set_fact: log_file="{{remote_path}}{{output_file}}.{{ec2_region}}.{{index}}"
     - set_fact: range_start={{1 + (populate|int) * (index|int) }}
     - set_fact: range_end={{(populate|int) * (1 + (index|int)) }}
     - set_fact: range="{{range_start}}..{{range_end}}"
     - debug: msg="client key range {{range}}"
```
Comment (tzach) [inline stress.yaml:47]: @glommer as far as you know, do we need to set different population per client?
@tgrabiec did you do so in yours Scylla/Cassandra benchmarks?

Comment (tgrabiec) [inline stress.yaml:47]: 2015-11-05 7:59 GMT+01:00 Tzach Livyatan notifications@github.com:

> In stress.yaml
> https://github.com/scylladb/cassandra-test-and-deploy/pull/21#discussion_r43981507
> :
> 
> > @@ -27,19 +27,32 @@
> >      - set_fact: index={{ec2_tag_index}}
> >        when: index is not defined
> > -    - set_fact: taskset="taskset -c {{load_cpus}}"
> > -      when: load_cpus is defined
> >   +
> > -    - set_fact: taskset=""
> > -      when: load_cpus is not defined
> >   +
> > -    - debug: msg="setting network IRQ affinity to CPU0"
> > -      when: load_cpus is defined
> >   +
> > -    - shell: for irq in `cat  /proc/interrupts | grep eth0 | cut -d":" -f1`; do echo 1 > /proc/irq/$irq/smp_affinity; done
> > -      when: load_cpus is defined
> > -      sudo: yes
> >   +
> >   - set_fact: log_file="{{remote_path}}{{output_file}}.{{ec2_region}}.{{index}}"
> >   - set_fact: range_start={{1 + (populate|int) \* (index|int) }}
> >   - set_fact: range_end={{(populate|int) \* (1 + (index|int)) }}
> >   - set_fact: range="{{range_start}}..{{range_end}}"
> >   - debug: msg="client key range {{range}}"
> 
> @glommer https://github.com/glommer as far as you know, do we need to
> set different population per client?
> @tgrabiec https://github.com/tgrabiec did you do so in yours
> Scylla/Cassandra benchmarks?

In my benchmarks I used the same population for all clients. It depends
what kind of workload you want to test.

**Comment by tzach** [comment]
> In my benchmarks I used the same population for all clients. It depends
> what kind of workload you want to test.

This feature make the benchmark more complex, hard to reason about.
It might be more realistic, but if no one else is doing that, we shouldn't either

**Comment by tgrabiec** [comment]
@tzach which feature, what's complex?

File: stress.yaml (hunk 2 above)
Comment (glommer) [inline stress.yaml:34]: Maybe. I had runs with and without it, so from a dev PoV, it is definitely nice to be able to take this in and out.

But as for the finished thing, if we establish that this is always the best way, we can certainly make it a property of the node.

**Pull request merged**